runtime/autoload/health.vim

```diff
@@ -43,19 +43,23 @@ function! health#check(plugin_names) abort
     redraw|echo 'Running healthchecks...'
     for c in healthchecks
       let output = ''
-      call append('$', split(printf("\n%s\n%s", c, repeat('=',72)), "\n"))
+      let disp = c[2] != '' ? c[2] : c[1]
+      call append('$', split(printf("\n%s: %s\n%s", c[0], disp, repeat('=',72)), "\n"))
       try
-        let output = "\n\n".execute('call '.c.'()')
+        let output = "\n\n".execute('call '.c[1].'()')
+        if output =~# 'HEALTHCHECK_ERROR:'
+          throw matchlist(output, 'HEALTHCHECK_ERROR:\(.*$\)')[1]
```

Comment:
instead of this , let's require Lua healthchecks to return a structure (string, or even a map of some sort) instead of this output scraping. The output scraping was a design mistake.

Reply:
I refactored health.vim to use variable scope, now every report function instead of echoing the formatted parameters appends to an `output` list. So now in the lua side things are cleaner the report functions are wrapped to look nicer. Let me know what you think.